.sandbox/provision/lifecycle-asg/common.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+
+: ${MAX_RETRIES:=30}
+: ${VOLUME_SIZE:=10}
+: ${DEVICE_NAME:="/dev/xvdf"}
+: ${SANDBOX_NAME_TAG:="SandboxManaged-SandboxName"}
+: ${SANDBOX_ID_TAG:="SandboxManaged-SandboxID"}
+: ${SANDBOX_ASG_TAG:="sandbox-asg"}
+
+function fatal() {
+  echo "$@" >&2
+  exit 1
+}
+
+function get_volume_id() {
+    local volume
+    volume="$(aws ec2 describe-volumes --filters Name=tag:"$SANDBOX_ID_TAG",Values="$SANDBOX_ID")"
+    jq -cMr '.Volumes[0].VolumeId' <<< "$volume"
+}
+
+function create_volume_if_needed() {
+    local volume_id="$(get_volume_id)"
+    if [[ "$volume_id" -eq null ]]; then
+        volume="$(aws ec2 create-volume --size "$VOLUME_SIZE" --availability-zone "$AVAILABILITY_ZONE" --tag-specification "ResourceType=volume,Tags=[{Key=$SANDBOX_ID_TAG,Value="$SANDBOX_ID"}]")"
+        volume_id="$(echo "$volume" | jq -cMr '.VolumeId')"
+    fi
+    echo "$volume_id"
+}
+
+function delete_volume() {
+    local volume_id="$(get_volume_id)"
+    [[ "$volume_id" = "null" ]] || aws ec2 delete-volume --volume-id "$volume_id"
+}
+
+function get_instance_id() {
+    instance="$(aws ec2 describe-instances --filters Name=tag:"$SANDBOX_ID_TAG",Values="$SANDBOX_ID" Name=instance-state-name,Values=running)"
+    jq -cMr '.Reservations[0].Instances[0].InstanceId' <<< "$instance"
```

Comment:
This is something I tested, we can use `--query '.Reservations[0].Instances[0].InstanceId' --output text` without relying on `jq`.

Reply:
As I mentioned in another comment, I prefer to use a unified solution for response parsing since we could not fully get rid of `jq`. `--output text` across the scripts. 

Your suggestion does work and we could use it. The only difference is the empty value is `None` instead of `null` if nothing returns.